src/config/backends/splash.ts

```diff
@@ -0,0 +1,187 @@
+import * as Bluebird from 'bluebird';
+import * as _ from 'lodash';
+import { fs } from 'mz';
+import * as path from 'path';
+
+import * as constants from '../../lib/constants';
+import log from '../../lib/supervisor-console';
+import * as logger from '../../logger';
+import {
+	bootMountPoint,
+	ConfigBackend,
+	ConfigOptions,
+	remountAndWriteAtomic,
+} from './backend';
+
+export class SplashImage extends ConfigBackend {
+	private static readonly BASEPATH = path.join(bootMountPoint, 'splash');
+	private static readonly DEFAULT = path.join(
+		SplashImage.BASEPATH,
+		'balena-logo-default.png',
+	);
+	private static readonly PATHS = [
+		'balena-logo.png',
+		'resin-logo.png',
+	].map((logo) => path.join(SplashImage.BASEPATH, logo));
+	private static readonly PREFIX = `${constants.hostConfigVarPrefix}SPLASH_`;
+	private static readonly CONFIGS = ['image'];
+
+	// Check the header of the image for a png
+	// https://github.com/sindresorhus/is-png/blob/master/index.js
+	private isPng(buffer: Buffer) {
+		if (!buffer || buffer.length < 8) {
+			return false;
+		}
+
+		return (
+			buffer[0] === 0x89 &&
+			buffer[1] === 0x50 &&
+			buffer[2] === 0x4e &&
+			buffer[3] === 0x47 &&
+			buffer[4] === 0x0d &&
+			buffer[5] === 0x0a &&
+			buffer[6] === 0x1a &&
+			buffer[7] === 0x0a
+		);
+	}
+
+	// Returns the base64 contents of the splash image
+	private async readSplashImage(where?: string): Promise<string> {
+		const [
+			// If no logo is defined, assume the file `balena-logo.png`
+			defaultPath = path.join(SplashImage.BASEPATH, 'balena-logo.png'),
+		] = await Bluebird.filter(SplashImage.PATHS, (logo) => fs.exists(logo));
+
+		// Read from defaultPath unless where is defined
+		where = where ?? defaultPath;
+
+		// read the image file...
+		try {
+			const buffer = await fs.readFile(where);
+
+			return buffer.toString('base64');
+		} catch (err) {
+			log.error('Failed to read splash image.', err);
+		}
+
+		return '';
```

Comment:
I'm in favour of throwing an exception when we fail to read the splash image. This is because the backend has a few places where it does `!_.isEmpty(image)` and I those can be avoided if we just throw an exception and catch it where ever needed.